src/gt4py/next/program_processors/runners/dace_fieldview/gtir_sdfg.py

```diff
@@ -100,22 +100,40 @@ class SDFGBuilder(DataflowBuilder, Protocol):
     """Visitor interface available to GTIR-primitive translators."""
 
     @abc.abstractmethod
-    def make_field(
-        self, data_node: dace.nodes.AccessNode, data_type: ts.FieldType | ts.ScalarType
-    ) -> gtir_builtin_translators.FieldopData:
-        """Retrieve the field data descriptor including the domain offset information."""
+    def get_sdfg(self) -> dace.SDFG:
+        """Retrieve the SDFG being built."""
         ...
 
     @abc.abstractmethod
     def get_symbol_type(self, symbol_name: str) -> ts.DataType:
         """Retrieve the GT4Py type of a symbol used in the SDFG."""
         ...
 
+    @abc.abstractmethod
+    def make_field(
+        self, data_node: dace.nodes.AccessNode, data_type: ts.FieldType | ts.ScalarType
+    ) -> gtir_builtin_translators.FieldopData:
+        """Retrieve the field data descriptor including the domain offset information."""
+        ...
+
     @abc.abstractmethod
     def visit(self, node: concepts.RootNode, **kwargs: Any) -> Any:
         """Visit a node of the GT4Py IR."""
         ...
 
+    def add_temp_array(
+        self, shape: Sequence[Any], dtype: dace.dtypes.typeclass, **kwargs: Any
+    ) -> Tuple[str, dace.dtypes.Array]:
+        """Wrapper of `dace.SDFG.add_temp_transient()`."""
+        return self.get_sdfg().add_temp_transient(shape, dtype, **kwargs)
+
+    def add_temp_scalar(
+        self, dtype: dace.dtypes.typeclass, name: Optional[str] = None, **kwargs: Any
+    ) -> Tuple[str, dace.dtypes.Scalar]:
+        """Helper function analogue of `dace.SDFG.add_temp_transient()`, but for scalars."""
+        name = name or self.get_sdfg().temp_data_name()
+        return self.get_sdfg().add_scalar(name, dtype, **kwargs)
```

Comment:
Currently you can not do that, but I think it should be possible to write:
```python
sdfg.add_scalar(name=None, ...)
```
and this implies `find_new_name=True`.

Reply:
Yes, that would be simpler.